### PR TITLE
Add can_forward to index + other bits

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/index/api/src/main.rs [hunk 1]
```diff
@@ -1,7 +1,9 @@
 use candid_gen::generate_candid_method;
 
 fn main() {
+    generate_candid_method!(index, allocated_bucket, query);
```
Comment (hpeebles) [inline backend/canisters/index/api/src/main.rs:4]: Why add this back in?

File: backend/canisters/index/impl/src/queries/can_forward.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+use canister_api_macros::trace;
+use ic_cdk_macros::query;
+use index_canister::{
+    can_forward::{Response::*, *},
+    ProjectedAllowance,
+};
+
+#[query]
+#[trace]
+fn reference_counts(args: Args) -> Response {
```
Comment (hpeebles) [inline backend/canisters/index/impl/src/queries/can_forward.rs:11]: Name is wrong...

File: backend/canisters/index/api/src/queries/allocated_bucket.rs [hunk 3]
```diff
@@ -1,48 +1,28 @@
...
-            v2::Response::Success(r) => Response::Success(Result {
-                canister_id: r.canister_id,
-                chunk_size: r.chunk_size,
-            }),
-            v2::Response::AllowanceExceeded(_) => Response::AllowanceReached,
-            v2::Response::UserNotFound => Response::UserNotFound,
-            v2::Response::BucketUnavailable => Response::BucketUnavailable,
-        }
-    }
+    pub byte_limit: u64,
```
Comment (hpeebles) [inline backend/canisters/index/api/src/queries/allocated_bucket.rs:24]: These fields should go, right?

Comment (megrogan) [reply backend/canisters/index/api/src/queries/allocated_bucket.rs]: yes

**Review by hpeebles — APPROVED** [review]

**Pull request merged**